packages/make-promise/src/makePromise.js

```diff
@@ -28,7 +28,11 @@ export default function makePromise() {
       delete p.domain;
     }
   }
-  // TODO: Retire name 'rej' as it looks too much like 'res'.
-  return harden({ p, res, rej, reject: rej });
+  return harden({ promise: p, resolve: res, reject: rej });
 }
 harden(makePromise);
+
+export function isPromise(maybePromise) {
+  return Promise.resolve(maybePromise) === maybePromise;
```

Comment:
question (maybe for @michaelfig or @erights) does this need to be `HandledPromise.resolve`? I *think* the answer is no, but I want to double check.

Reply:
The answer is no.

Reply:
Agreed that either is correct. They are equivalent for this test. However, they are not equivalent in general and we should generally encourage our own code (at least) to generally use `HandledPromise.resolve` rather than `Promise.resolve`. The difference is: when given a non-promise, does it wrap it in a handled or non-handled promise? Every unhandled promise creates loss of pipelining and reordering that will not be equivalent to what we're proposing. The shim can't fix this. But the HandledPromise in the shim interacting with other handled promises does do exactly what we're proposing.

I would prefer `HandledPromise` here unless there's a reason not to.

Reply:
done